content/posts/2021-05-airdrop-anywhere-part-2.md

```diff
@@ -0,0 +1,167 @@
+---
+title: "AirDrop Anywhere - Part 2 - Writing some code"
+date: 2021-05-04T15:00:00Z
+tags: [.net, networking, airdrop, apple]
+images: [img/airdrop-anywhere-cover.jpg]
+---
+
+> This is part 2 of a series of posts:
+> - [Part 1 - Introduction](/posts/2021-05-airdrop-anywhere-part-1)
+> - [Part 2 - Writing some code](/posts/2021-05-airdrop-anywhere-part-2)
+> - [GitHub (mDNS bits)](https://github.com/deanward81/AirDropAnywhere/tree/2021-05-04-mdns)
+> - [GitHub (latest)](https://github.com/deanward81/AirDropAnywhere/tree/main) - **NOTE** still work in progress!
+
+
+[Last time](/posts/2021-04-airdrop-anywhere-part-1) we broke down the problem of implementing AirDrop so that we can support sending and receiving files from devices that do not natively support AirDrop. After some to and fro we landed on an implementation that involves creating a "proxy" running on an Apple device (which supports AWDL natively) or Linux with [OWL](https://owlink.org).
+
+## Pre-requisites
+
+### Project structure
+I've decided to structure all of this as a "Core" project that'll contain the bulk of the AirDrop implementation. This will be able to be plugged into any .NET Core application in order to provide AirDrop services and each application can provide an implementation of an interface in order to handle file sending and receiving. Our project hierarchy will look like this:
+
+```
+|---AirDropAnywhere.Core
+    |
+    |--AirDropAnywhere.Cli
+    |
+    |--AirdropAnywhere.Web
+```
```

Comment:
I get the intent here, but visually it looks like .Core contains the other 2, rather than the other way around - maybe a different dependency diagram would be clearer?